tool/diagnostic_update_check.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:crypto/crypto.dart' as crypto;
+import 'package:http/http.dart' as http;
+
+void main() async {
+  print('Verifying that diagnostic messages file is up to date...');
+
+  // Uncomment once linter rules are added to website
+  // if (!await checkAsset(
+  //     'https://raw.githubusercontent.com/dart-lang/linter/gh-pages/lints/machine/rules.json',
+  //     '../src/_data/linter_rules.json')) {
+  //   print('The linter rules file is outdated.');
+  //   exit(1);
+  // }
+
+  if (!await checkAsset(
+      'https://raw.githubusercontent.com/dart-lang/sdk/master/pkg/analyzer/tool/diagnostics/diagnostics.md',
+      '../src/tools/diagnostic-messages.md')) {
+    print('The diagnostic messages file is outdated.');
+    exit(1);
+  }
+
+  print('The diagnostic messages file is up to date.');
+}
+
+Future<bool> checkAsset(final String url, final String path) async {
+  final remote = await http.get(Uri.parse(url));
+  if (remote.statusCode == 200) {
+    final local = File(path);
+    final remoteBytes = remote.bodyBytes;
+    final localBytes = local.readAsBytesSync();
+    if (remoteBytes.length == localBytes.length) {
+      final remoteHash = crypto.sha256.convert(remoteBytes).toString();
+      final localHash = crypto.sha256.convert(localBytes).toString();
```

Comment:
Is there any value in comparing the sha's rather than just comparing the contents directly?

Reply:
I was just following the `pub-dev` implementation for their comparison checks, but after some basic benchmarking, comparing the contents is faster and more straightforward anyway. Thanks for the suggestion!